Gauge/Benchmark.hs

```diff
@@ -182,12 +182,20 @@ pureFunc reduce f0 x0 = toBenchmarkable (go f0 x0)
 
 -- | Apply an argument to a function, and evaluate the result to weak
 -- head normal form (WHNF).
+-- Argument evaluation cost is excluded from the results, see `nf`.
 whnf :: (a -> b) -> a -> Benchmarkable
 whnf = pureFunc id
 {-# INLINE whnf #-}
 
 -- | Apply an argument to a function, and evaluate the result to
 -- normal form (NF).
+-- By default, this function call will be measured several times.
+-- Argument evaluation happens only once in the first iteration.
+-- In later iterations, the previously evaluated argument value is used.
+-- By default, we discard the measurement of the first iteration
+-- so the argument evaluation cost gets excluded from the results.
```

Comment:
I tried to reword a bit, see if this sounds like an improvement:

By default, multiple iterations are performed to measure the cost of function application. Since the argument is pure it is evaluated only once in the first iteration and subsequent iterations end up using the evaluated argument. To exclude the cost of one time argument evaluation from the final results `gauge` discards the first iteration measurement.

Reply:
OK fine.